src/watertap_contrib/reflo/property_models/fo_draw_solution_properties.py

```diff
@@ -0,0 +1,815 @@
+            self.params.phase_list, rule=rule_flow_vol_phase
+        )
+
+    def _conc_mass_phase_comp(self):
+        self.conc_mass_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize=100,
+            bounds=(0.0, 1e6),
+            units=pyunits.kg * pyunits.m**-3,
+            doc="Mass concentration",
+        )
+
+        def rule_conc_mass_phase_comp(b, p, j):
+            return (
+                b.conc_mass_phase_comp[p, j]
+                == b.dens_mass_phase[p] * b.mass_frac_phase_comp[p, j]
+            )
+
+        self.eq_conc_mass_phase_comp = Constraint(
+            self.params.phase_list,
+            self.params.component_list,
+            rule=rule_conc_mass_phase_comp,
+        )
+
+    def _pressure_osm_phase(self):
+        self.pressure_osm_phase = Var(
+            self.params.phase_list,
+            initialize=1e6,
+            bounds=(-1e6, 1e8),
+            units=pyunits.Pa,
+            doc="Osmotic pressure",
+        )
+
+        def rule_pressure_osm_phase(
+            b, p
+        ):  # osmotic pressure, derived from experimental data
+            s = b.mass_frac_phase_comp[p, "DrawSolution"]
+            pressure_osm_phase = (
+                b.params.osm_coeff_param_0
+                + b.params.osm_coeff_param_1 * s
+                + b.params.osm_coeff_param_2 * s**2
+                + b.params.osm_coeff_param_3 * s**3
+            )
+            return b.pressure_osm_phase[p] == pressure_osm_phase
+
+        self.eq_pressure_osm_phase = Constraint(
+            self.params.phase_list, rule=rule_pressure_osm_phase
+        )
+
+    def _cp_mass_phase(self):
+        self.cp_mass_phase = Var(
+            self.params.phase_list,
+            initialize=4e3,
+            bounds=(0.0, 1e8),
+            units=pyunits.J / pyunits.kg / pyunits.K,
+            doc="Specific heat capacity",
+        )
+
+        def rule_cp_mass_phase(
+            b, p
+        ):  # specific heat, equation derived from experimental data
+            s = b.mass_frac_phase_comp[p, "DrawSolution"]
+            cp_mass_phase = (
+                b.params.cp_phase_param_A0
+                + b.params.cp_phase_param_A1 * s
+                + b.params.cp_phase_param_A2 * s**2
+            )
+
+            return b.cp_mass_phase[p] == cp_mass_phase
+
+        self.eq_cp_mass_phase = Constraint(
+            self.params.phase_list, rule=rule_cp_mass_phase
+        )
+
+    def _heat_separation_phase(self):
+        self.heat_separation_phase = Var(
+            self.params.phase_list,
+            initialize=1e6,
+            bounds=(-1e10, 1e10),
+            units=pyunits.J / pyunits.s,
+            doc="Heat of liquid separation",
+        )
+        self.liquid_separation = Var(
+            initialize=0.1,
+            units=pyunits.dimensionless,
+            doc="Indication of whether liquid separation will happen",
+        )
```

Comment:
I see why this may have been removed from the ```define_custom_properties``` dict above, since it looks like this is meant to be some kind of boolean indicator and I imagine that is tricky to deal with in a property package. I saw you had a ```Binary``` import and probably discovered that IPOPT can't handle it (i.e., it is treated as a float), but if you would like to include this and need this functionality, you can try ```Disjunct```, otherwise might want to remove this entire ```_heat_separation_phase```.

Reply:
@Zhuoran29 curious if you ever made ust of the ```Disjunct``` for this model/flowsheet? If so, consider converting this PR to a draft until you are finished with the feature.

Reply:
@kurbansitterley I gave it a try but it didn't work out. I only found a few examples using pyomo disjunct and those examples didn't work themselves on my machine. 
The current approach of using `Expr_if` is working fine, should we stick to this approach before I educate myself enough about this feature?

Reply:
That's ok, I am not very familiar with the ```Disjunct``` but just wanted to check if you were using it before we merged this PR. 

If the situation can be handled with an if statement, and ```Expr_if``` is working, I don't see any reason to force it. I imagine the ```Disjunct``` would be needed if this decision were "optimizable," but I don't think it is.

Reply:
Yes, `Disjunct` is a more generalized feature than `Expr_if` and is more applicable when optimizable decision is involved. In this flowsheet the indicator/decision is fixed based on the configuration and shouldn't be mutable.